### PR TITLE
Add 404 error page and improve sitemap

### PR DESCRIPTION
**Pull request by sudo42**
- A `404` error page is added
- An option is added to exclude pages from the sitemap

### CODE REVIEW

File: layouts/sitemap.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    <loc>{{ .Permalink }}</loc>{{ if not .Lastmod.IsZero }}
+    <lastmod>{{ safeHTML ( .Lastmod.Format "2006-01-02T15:04:05-07:00" ) }}</lastmod>{{ end }}{{ with .Sitemap.ChangeFreq }}
+    <changefreq>{{ . }}</changefreq>{{ end }}{{ if ge .Sitemap.Priority 0.0 }}
+    <priority>{{ .Sitemap.Priority }}</priority>{{ end }}{{ if .IsTranslated }}{{ range .Translations }}
+    <xhtml:link
+                rel="alternate"
+                hreflang="{{ .Lang }}"
+                href="{{ .Permalink }}"
+                />{{ end }}
+    <xhtml:link
```
Comment (Paradyx) [inline layouts/sitemap.xml:14]: Why is this a duplicate of line 9 onwards?

**Review by Paradyx — COMMENTED** [review]
I don't quite get the code. Is this a duplicate?

File: layouts/sitemap.xml (hunk 1 above)
Comment (sudo42) [reply layouts/sitemap.xml:14]: this is used to list links to the same page in different languages (should they exist)

Comment (Paradyx) [reply layouts/sitemap.xml:14]: Still not quite getting it. 
First a loop for all translated pages and then the default page? What happens if the page is not translated (`.isTranslated == false`) from my understanding no link is added it that case.

Comment (sudo42) [reply layouts/sitemap.xml:14]: The "normal" URL is stored in `<loc>` and is added for all pages (except those, which exclude themselves).

Comment (Paradyx) [reply layouts/sitemap.xml:14]: Resolving this, although I am still skeptical

**Pull request merged**